staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -21,11 +21,17 @@ import (
 )
 
 var (
+	// DefaultMutableVersionedFeatureGate is DefaultMutableFeatureGate with access to settings specific to the emuation version.
+	// Only top-level commands/options setup and the k8s.io/component-base/featuregate/testing package should make use of this.
+	DefaultMutableVersionedFeatureGate featuregate.MutableVersionedFeatureGate = featuregate.NewFeatureGate()
+	// DefaultVersionedFeatureGate is a shared global FeatureGate with emulation version.
+	// Top-level commands/options setup that needs to modify this feature gate should use DefaultMutableVersionedFeatureGate.
+	DefaultVersionedFeatureGate featuregate.VersionedFeatureGate = DefaultMutableVersionedFeatureGate
```

Comment:
the existing DefaultMutableFeatureGate / DefaultFeatureGate split was already a little confusing (I can say that and only throw myself under the bus, I added it in a stop-gap fix in https://github.com/kubernetes/kubernetes/pull/71302).

Doubling the globals to have two mutable global gates and two non-mutable global gates is sort of going fractal... when should code use DefaultVersionedFeatureGate and when should it use DefaultFeatureGate? having two global vars that should *always* (?) refer to the same underlying instance is really confusing